### PR TITLE
Paginate Next Reward Payment

### PR DESCRIPTION
**Pull request by jholdstock**
Pagination for "Next Reward Payment" on the homepage.

### CODE REVIEW

File: gui/cache.go [hunk 1]
```diff
@@ -94,6 +95,13 @@ func (c *Cache) getMinedWork() []minedWork {
 
 // updateQuotas refreshes the cached list of pending dividend payments.
 func (c *Cache) updateQuotas(quotas []*pool.Quota) {
+
+	// Sort list so the largest percentages will be shown first.
+	sort.Slice(quotas, func(i, j int) bool {
+		larger := quotas[i].Percentage.Cmp(quotas[j].Percentage) > 0
```
Comment (dnldd) [inline gui/cache.go:102]: simpler to do `return quotas[i].Percentage.Cmp(quotas[j].Percentage) > 0`

File: gui/cache.go [hunk 2]
```diff
@@ -48,8 +48,8 @@ type Cache struct {
 	blockExplorerURL string
 	minedWork        []minedWork
 	minedWorkMtx     sync.RWMutex
-	workQuotas       []workQuota
-	workQuotasMtx    sync.RWMutex
+	dividends        []dividend
```
Comment (dnldd) [inline gui/cache.go:51]: Hmm not sure about this change considering the values being reported here are not actual amounts being paid to participating accounts but percentages of the work contributed so far towards finding the next block. Maybe `work quota` is also not descriptive or simple enough, but using `dividend` here is misleading.

File: gui/cache.go (hunk 1 above)
Comment (jholdstock) [reply gui/cache.go]: agreed

File: gui/cache.go (hunk 2 above)
Comment (jholdstock) [reply gui/cache.go]: Renamed to RewardQuota as discussed on Matrix

File: gui/index.go [hunk 3]
```diff
@@ -42,6 +41,12 @@ func (ui *GUI) renderIndex(w http.ResponseWriter, r *http.Request, modalError st
 		}
 	}
 
+	// Get the next reward payment percentages (max 10).
+	rewardQuotas := ui.cache.getRewardQuotas()
```
[the remark this answers is not in the document]
Comment (jholdstock) [reply gui/index.go:45]: This is for the initial render. When the page is first loaded we only want to display the top ten (ie. the first page). 

When the pagination Javascript loads, clickable buttons are displayed which allow the user to load additional data on the next pages. If the Javascript does not load, only the first 10 results will be visible (hence #181).

If this is not limited, it is a potential DoS vector. We had a similar problem with VSP where information about 100s of tickets was passed into a template which caused the templating engine to take 10+ seconds to render.

[the remark this answers is not in the document]
Comment (dnldd) [reply gui/index.go:45]: Gotcha, thanks.

**Pull request merged**